jbcli/jbcli/utils/reload.py

```diff
@@ -0,0 +1,41 @@
+"""Handles watcher-related tasks for app reloading
+"""
+import os
+import time
+
+import click
+from requests import get
+
+
+def create_browser_instance():
+    """Create proxy browser instance for hot reloading
+    """
+    cmd = 'npx browser-sync start --proxy="localhost:8000"'
+    os.system(cmd)
+
+
+def refresh_browser(timeout=None):
+    """Refreshes browser-sync browser instance if
+    Django server is ready
+
+    :param timeout: Optional timeout duration before checking
+    server status
+    """
+    click.echo('Checking server status...')
+
+    if timeout:
+        time.sleep(timeout)
+
+    def attempt_refresh():
+        try:
+            response = get('http://localhost:8000/health_check')
+            if response and response.status_code == 200:
+                click.echo('Refreshing browser...')
+                cmd = "npx browser-sync reload"
+                os.system(cmd)
+        except:
```

Comment:
I would prefer it if we had a more specific exception handler here. What exceptions are you expecting to be caught here? Something from `os.system`, or from the `get` call? I worry that if we get some unexpected error, this code will just loop forever and never print anything to the console indicating what kind of error happened.

So I have two suggestions:

1. only put the `try/except` around the expressions that you expect to raise specific errors
2. be more specific about what exception types you want to catch

Reply:
Ah, that absolutely makes sense. Will address.